### DISCUSSION TITLE
Redux state reset on route change

### DISCUSSION THREAD

**Discussion by prasanthbattula** [vercel/next.js#28839] [Help] [answered]
I am changing the route with **router.push**. After route change the redux state is resetting to initial state. I don't want this.
Could you please help me.

Thank You.

**Comment by bratsos** [comment, marked as answer]
If this is happening while changing pages (`/foo` -> `/bar`), it's expected, since the shallow routing [only works for same page urls](https://nextjs.org/docs/routing/shallow-routing#caveats). If you're using `router.push` for different pages, that will cause your whole app to be "remounted" thus resetting the state of your redux store.

One possible solution to that would be to save the state of your redux store on localstorage and rehydrate it after the new page loads (using a library like this [redux-persist](https://github.com/rt2zz/redux-persist)).
